### ISSUE TITLE
`extra-1.0` incompatible with GHC 7.0

### ISSUE THREAD

**Issue by hvr** [ndmitchell/extra#3]
Please either make it compatible with GHC 7.0 again or add a `base >= 4.4` constraint for the next release...

```
Building extra-1.0...
Preprocessing library extra-1.0...
[ 1 of 15] Compiling System.Info.Extra ( src/System/Info/Extra.hs, dist/build/System/Info/Extra.o )
[ 2 of 15] Compiling Numeric.Extra    ( src/Numeric/Extra.hs, dist/build/Numeric/Extra.o )
[ 3 of 15] Compiling Data.Tuple.Extra ( src/Data/Tuple/Extra.hs, dist/build/Data/Tuple/Extra.o )
[ 4 of 15] Compiling Data.List.Extra  ( src/Data/List/Extra.hs, dist/build/Data/List/Extra.o )
[ 5 of 15] Compiling Data.IORef.Extra ( src/Data/IORef/Extra.hs, dist/build/Data/IORef/Extra.o )
[ 6 of 15] Compiling Data.Either.Extra ( src/Data/Either/Extra.hs, dist/build/Data/Either/Extra.o )
[ 7 of 15] Compiling Control.Monad.Extra ( src/Control/Monad/Extra.hs, dist/build/Control/Monad/Extra.o )
[ 8 of 15] Compiling System.Directory.Extra ( src/System/Directory/Extra.hs, dist/build/System/Directory/Extra.o )
[ 9 of 15] Compiling Control.Exception.Extra ( src/Control/Exception/Extra.hs, dist/build/Control/Exception/Extra.o )
[10 of 15] Compiling System.Environment.Extra ( src/System/Environment/Extra.hs, dist/build/System/Environment/Extra.o )
[11 of 15] Compiling System.Time.Extra ( src/System/Time/Extra.hs, dist/build/System/Time/Extra.o )

src/System/Time/Extra.hs:70:30: Not in scope: `forkIOWithUnmask'
```

**Comment by ndmitchell** [comment]
I'm surprised you decided to go back as far as 7.0 - I know Standard Chartered is still on 7.2, but I thought we were miles behind the curve. Anyway, I'll certainly include this in the next release, but that might not be for a few weeks (my dev machine blew up yesterday).

**Comment by hvr** [comment]
@ndmitchell it just showed up while I was throwing that build-matrix-builder at `extra`, and GHC 7.0.4 just happens to be part of the test-matrix. (if you're curious about some example reports: https://ghc.haskell.org/~hvr/buildreports/0INDEX.html)

It's totally fine if future `extra` version drop support for GHC 7.0. In that case, by placing a `base >= 4.4` just keeps those few GHC 7.0 users from having `cabal install extra` broken suddenly  when you release a new `extra` package...

**Comment by ndmitchell** [comment]
Bound included, and a new extra-1.0.1 release made. Let me know if any of my other packages trip up your testing. While my base bounds aren't always correct, my tested-with lines are, and if you start relying on the base bounds, I can update my checking script to demand them too.
